src/assimp/SceneConverter.cpp

```diff
@@ -14,6 +14,18 @@ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLI
 
 using namespace vsgXchange;
 
+namespace
+{
+    [[always_inline, msvc::forceinline]]
```

Comment:
If we are concerned enough about performance to add compiler directives for inlining then I think it's an indication that the code might be best served by having two methods - an inline one for single color value changes, and one for arrays.  

The later will be the main performance bottleneck and by having an array version of convertColor one could move the comparison test outside the inner for loop.

Reply:
I went for this approach as it seemed like the most concise way to encourage the compiler to hoist the `if` outside of the loop. Duplicating the function with a specialisation that takes an array (or something more generic like iterators which would mean it had to be a template) would also achieve the same effect. I didn't want to duplicate anything as this seemed like a function that could get more complicated in the future (as sRGB and linear aren't the only colour spaces in the wild - even before modern HDR formats became common, Adobe RGB was widely used) or might warrant moving to `color.h` in the main VSG repo, or it might want switching from `vsg::Options::ColorSpace` to `VkColorSpaceKHR` (and then it'd need to handle the sixteen values that enum can have and grow to support any more added by future Vulkan versions and extensions). Those seemed like sensible things to discuss before dealing too much with the implementation details of the current approach.

Reply:
I've had a play with Godbolt, and determined that GCC and Clang both want to inline `convertColor` if it's called in a loop even without any hinting. However, MSVC won't do it without `[[msvc::forceinline]]` or an equivalent. `inline` doesn't make it inline it, but that's to be expected as it's been a long time since compilers used it for that and instead it just means that the linker won't error out if there are duplicate identical definitions in multiple translation units. That means that the implementation will need either writing out twice or a `[[msvc::forceinline]]`.